.github/workflows/test.yml

```diff
@@ -10,8 +10,18 @@ jobs:
       fail-fast: false
       max-parallel: 5
       matrix:
-        python-version: ['3.8', '3.9', '3.10', '3.11', 'pypy-3.10']
-        django-version: ['2.2', '3.2', '4.0', '4.1', '4.2']
+        python-version:
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
+        - 'pypy-3.10'
+        django-version:
+        - '3.2'
+        - '4.1'
+        - '4.2'
+        - '5.0'
```

Comment:
breaking out like this reduces diff noise

Django 2.2 and 4.0 were left on the grid accidentally since d5ad2acb1c4c0397eb36a806e58399a9dd3ee286 removed support. I dropped them, and the below `exclude` rules relating to them.